### PR TITLE
[CALCITE-4976] Release Calcite 1.30.0

### PR DESCRIPTION
**Pull request by liyafan82**

### CODE REVIEW

File: site/_docs/history.md [hunk 1]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+contains contributions from 29 authors,
+and resolves 37 issues.
+
+Among others, it is worth highlighting the following.
+
+* [Babel parser support MySQL NULL-safe equal operator '<=>'](https://issues.apache.org/jira/browse/CALCITE-4980)
+* [Support SQL hints for temporal table join](https://issues.apache.org/jira/browse/CALCITE-4967)
+* [Fluent test fixtures so that dependent projects can write parser, validator and rules tests](https://issues.apache.org/jira/browse/CALCITE-4885">CALCITE-4885)
+* [Add functions `ARRAY_CONCAT`, `ARRAY_REVERSE`, `ARRAY_LENGTH` for BigQuery dialect](https://issues.apache.org/jira/browse/CALCITE-4822">CALCITE-4822)
+* [Vulnerability issues CVE-2021-27568 fixed](https://nvd.nist.gov/vuln/detail/CVE-2021-27568)
```
Comment (rubenada) [inline site/_docs/history.md:45]: minor: "issues" should be "issue"

File: site/_posts/2022-03-04-release-1.30.0.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+contains contributions from 29 authors,
+and resolves 37 issues.
+
+Among others, it is worth highlighting the following.
+
+* [Babel parser support MySQL NULL-safe equal operator '<=>'](https://issues.apache.org/jira/browse/CALCITE-4980)
+* [Support SQL hints for temporal table join](https://issues.apache.org/jira/browse/CALCITE-4967)
+* [Fluent test fixtures so that dependent projects can write parser, validator and rules tests](https://issues.apache.org/jira/browse/CALCITE-4885">CALCITE-4885)
+* [Add functions `ARRAY_CONCAT`, `ARRAY_REVERSE`, `ARRAY_LENGTH` for BigQuery dialect](https://issues.apache.org/jira/browse/CALCITE-4822">CALCITE-4822)
+* [Vulnerability issues CVE-2021-27568 fixed](https://nvd.nist.gov/vuln/detail/CVE-2021-27568)
```
Comment (rubenada) [inline site/_posts/2022-03-04-release-1.30.0.md:43]: minor: "issues" should be "issue"

File: site/_docs/history.md (hunk 1 above)
Comment (liyafan82) [reply site/_docs/history.md]: Revised. Thanks for your careful review.

File: site/_docs/howto.md [hunk 3]
```diff
@@ -693,8 +693,11 @@ Before you start:
   is starting and therefore `master` branch is in code freeze until further notice.
 * Set up signing keys as described above.
 * Make sure you are using JDK 8 (not 9 or 10).
-* Make sure `master` branch and `site` branch are in sync, i.e. there is no commit on `site` that has not
-  been applied also to `master`. If you spot missing commits then port them to `master`.
+* Make sure `master` branch and `site` branch are in sync, i.e. there is no
+  commit on `site` that has not been applied also to `master`.
+  We are talking about the commit content, you need to pay attention to the commit message
+  and change, not hash: it is normal to have the same change in site and
```
Comment (rubenada) [inline site/_docs/howto.md:699]: "site" and "master" are missing the backtick in the last sentence. Also please keep at the end the sentence: "If you spot missing commits then port them to `master`."

File: site/_posts/2022-03-04-release-1.30.0.md (hunk 2 above)
Comment (liyafan82) [reply site/_posts/2022-03-04-release-1.30.0.md]: Revised. Please check. Thanks.

File: site/_docs/howto.md (hunk 3 above)
Comment (liyafan82) [reply site/_docs/howto.md]: Fixed. Thanks for your kind reminder.

File: site/_posts/2022-03-04-release-1.30.0.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+contains contributions from 29 authors,
+and resolves 37 issues.
+
+Among others, it is worth highlighting the following.
+
+* [Babel parser support MySQL NULL-safe equal operator '<=>'](https://issues.apache.org/jira/browse/CALCITE-4980)
+* [Support SQL hints for temporal table join](https://issues.apache.org/jira/browse/CALCITE-4967)
+* [Fluent test fixtures so that dependent projects can write parser, validator and rules tests](https://issues.apache.org/jira/browse/CALCITE-4885">CALCITE-4885)
+* [Add functions `ARRAY_CONCAT`, `ARRAY_REVERSE`, `ARRAY_LENGTH` for BigQuery dialect](https://issues.apache.org/jira/browse/CALCITE-4822">CALCITE-4822)
+* [Vulnerability issue CVE-2021-27568 fixed](https://nvd.nist.gov/vuln/detail/CVE-2021-27568)
```
Comment (zabetak) [inline site/_posts/2022-03-04-release-1.30.0.md:43]: See previous comments regarding this.

File: site/_docs/history.md [hunk 5]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+  Support Snapshot in `RelMdColumnOrigins`
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4872">CALCITE-4872</a>]
+  Add `UNKNOWN` value to enum `SqlTypeName`, distinct from the `NULL` type
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4702">CALCITE-4702</a>]
+  Error when executing query with GROUP BY constant via JDBC adapter
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4683">CALCITE-4683</a>]
+  IN-list converted to JOIN throws type mismatch exception
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4323">CALCITE-4323</a>]
+  If a view definition has an ORDER BY clause, retain the sort if the view is used in a query at top level
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4292">CALCITE-4292</a>]
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md]: Removed. Thanks.

File: site/_docs/history.md [hunk 6]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+  `RepeatUnion` containing a Correlate with a transientScan on its RHS causes NPE
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-3673">CALCITE-3673</a>]
+  `ListTransientTable` should not leave tables in the schema
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-3627">CALCITE-3627</a>]
+  Incorrect null semantic for ROW function
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-1794">CALCITE-1794</a>]
+  Expressions with numeric comparisons are not simplified when CAST is present
+
+#### Build and test suite
+{: #build-1-29-0}
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md]: Fixed. Sorry for my carelessness.

File: site/_docs/history.md [hunk 7]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+  Upgrade log4j2 version to 2.17.1
+
+#### Web site and documentation
+{: #site-1-30-0}
+
+* Site: Update PMC Chair
+* Site: Add external resources section in the community page
+* Site: Add "calcite-clj - Use Calcite with Clojure" in talks section
+* Site: Add Alessandro Solimando as committer
+* Site: Fix typo in howto.md
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md]: Removed. Thanks.

File: site/_docs/history.md [hunk 8]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+contains contributions from 29 authors,
+and resolves 37 issues.
+
+Among others, it is worth highlighting the following.
+
+* [Babel parser support MySQL NULL-safe equal operator '<=>'](https://issues.apache.org/jira/browse/CALCITE-4980)
+* [Support SQL hints for temporal table join](https://issues.apache.org/jira/browse/CALCITE-4967)
+* [Fluent test fixtures so that dependent projects can write parser, validator and rules tests](https://issues.apache.org/jira/browse/CALCITE-4885">CALCITE-4885)
+* [Add functions `ARRAY_CONCAT`, `ARRAY_REVERSE`, `ARRAY_LENGTH` for BigQuery dialect](https://issues.apache.org/jira/browse/CALCITE-4822">CALCITE-4822)
+* [Vulnerability issue CVE-2021-27568 fixed](https://nvd.nist.gov/vuln/detail/CVE-2021-27568)
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md]: Done. Thanks.

File: site/_docs/history.md [hunk 9]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+#### New features
+{: #new-features-1-30-0}
+
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4980">CALCITE-4980</a>]
+  Babel parser support MySQL NULL-safe equal operator '<=>'
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4967">CALCITE-4967</a>]
+  Support SQL hints for temporal table join
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4885">CALCITE-4885</a>]
+  Fluent test fixtures so that dependent projects can write parser, validator and rules tests
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4822">CALCITE-4822</a>]
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md]: Removed. Thanks.

File: site/_docs/history.md [hunk 10]
```diff
@@ -29,6 +29,162 @@ Downloads are available on the
...
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4901">CALCITE-4901</a>]
+  JDBC adapter incorrectly adds ORDER BY columns to the SELECT list of generated SQL query
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4877">CALCITE-4877</a>]
+  Support Snapshot in `RelMdColumnOrigins`
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4872">CALCITE-4872</a>]
+  Add `UNKNOWN` value to enum `SqlTypeName`, distinct from the `NULL` type
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4702">CALCITE-4702</a>]
+  Error when executing query with GROUP BY constant via JDBC adapter
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4683">CALCITE-4683</a>]
+  IN-list converted to JOIN throws type mismatch exception
```
[the remark this answers is not in the document]
Comment (liyafan82) [reply site/_docs/history.md:137]: Made another pass, and added backticks. Please check. Thanks.

File: site/_posts/2022-03-04-release-1.30.0.md (hunk 4 above)
Comment (liyafan82) [reply site/_posts/2022-03-04-release-1.30.0.md]: Added reference to JIRA. Thanks.

File: site/_docs/history.md [hunk 11]
```diff
@@ -29,6 +29,155 @@ Downloads are available on the
...
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4323">CALCITE-4323</a>]
+  If a view definition has an `ORDER BY` clause, retain the sort if the view is used in a query at top level
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-4054">CALCITE-4054</a>]
+  `RepeatUnion` containing a `Correlate` with a `transientScan` on its RHS causes NPE
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-3673">CALCITE-3673</a>]
+  `ListTransientTable` should not leave tables in the schema
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-3627">CALCITE-3627</a>]
+  Incorrect null semantic for ROW function
```
Comment (zabetak) [inline site/_docs/history.md:145]: Backticks for ROW.

Comment (liyafan82) [reply site/_docs/history.md]: Fixed. Thanks.

**Review by zabetak — APPROVED** [review]

**Comment by rubenada** [comment]
LGTM

**Comment by liyafan82** [comment]
@zabetak @rubenada Thanks a lot for your comments. If there are no more comments, I will merge this PR soon.

**Comment by liyafan82** [comment]
Merging. Thanks.

**Pull request merged**